packages/common/src/hardhat/tasks/artifacts.js

```diff
@@ -0,0 +1,206 @@
+
+  // Generate a unique list of artifacts and paths to them. Unique is necessary because there are some redundantly
+  // named contracts.
+  return uniqBy(
+    artifactPaths.map((artifactPath) => ({
+      contractName: path.basename(artifactPath).split(".")[0],
+      relativePath: `./${path.relative(path.dirname(relativeTo), artifactPath)}`,
+    })),
+    "contractName"
+  );
+}
+
+function getCorePath(hre, relativeTo) {
+  const artifactPath = hre.config.paths.artifacts ? path.join(hre.config.paths.artifacts, "../") : "./";
+  return `./${path.relative(relativeTo, artifactPath)}`;
+}
+
+function getAddressesMap(hre) {
+  // Generate a map of name => chain id => address.
+  const networksPath = path.join(getCorePath(hre, "./"), "networks");
+  const dirs = fs.readdirSync(networksPath);
+  const addresses = {};
+  for (const dir of dirs) {
+    const chainId = parseInt(dir.split(".")[0]);
+    const deployments = JSON.parse(fs.readFileSync(path.join(networksPath, dir), "utf8"));
+
+    // Loop over the deployments in the file and save each one.
+    for (const { contractName, address, deploymentName } of deployments) {
+      // If deploymentName isn't specified, use contractName.
+      const name = deploymentName ? deploymentName : contractName;
+      if (!addresses[name]) {
+        addresses[name] = {};
+      }
+      addresses[name][chainId] = address;
+    }
+  }
+  return addresses;
+}
+
+task("generate-contracts-frontend", "Generate typescipt for the contracts-frontend package")
+  .addParam("out", "browser ts output file", undefined, types.string)
+  .setAction(async function (taskArguments, hre) {
+    const { out } = taskArguments;
+    removeFileIfExists(out);
+
+    const artifacts = await getArtifactPathList(hre, "./");
+    const addresses = getAddressesMap(hre);
+
+    // Write Ethers contract types/factories export.
+    fs.appendFileSync(out, "export type {\n");
+    artifacts.forEach(({ contractName }) => {
+      if (fs.existsSync(`typechain/ethers/${contractName}.d.ts`))
+        fs.appendFileSync(out, `  ${contractName} as ${contractName}Ethers,\n`);
+    });
+    fs.appendFileSync(out, '} from "../typechain/ethers";\n');
+
+    fs.appendFileSync(out, "export {\n");
+    artifacts.forEach(({ contractName }) => {
+      if (fs.existsSync(`typechain/ethers/factories/${contractName}__factory.ts`))
+        fs.appendFileSync(out, `  ${contractName}__factory as ${contractName}Ethers__factory,\n`);
+    });
+    fs.appendFileSync(out, '} from "../typechain/ethers";\n');
+
+    // Write abi and bytecode for the browser file.
+    // Note: the idea behind writing the functions this way is to make them as optimized as possible for tree-shaking
+    // to remove any unused json files. In modern versions of webpack, this should allow absolutely _no_ artifact
+    // information that isn't needed to be pulled in.
+    // artifacts.forEach(({ contractName, relativePath }) =>
```

Comment:
is this meant to be commented out like this?

Reply:
Oops, will remove! Thanks for pointing that out!